mail_channel_sale_order_link/models/mail_channel.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2020 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import re
+
+from odoo import api, models
+
+
+class MailChannel(models.Model):
+    _inherit = "mail.channel"
+
+    @api.multi
+    @api.returns("mail.message", lambda value: value.id)
+    def message_post(self, message_type="notification", **kwargs):
+        # Split the message by spacing
+        keywords = re.split(r"\s+", kwargs.get("body"))
+        sale_orders = self.env["sale.order"].search([("name", "in", keywords)])
```

Comment:
What would be the impact to the performance in case there are a lot of strings in a message?  Would it make the performance better if we insert some pre-processing (e.g. identify keywords that start with a sales order prefix defined in ir.sequence)? I suppose the performance is OK with the current design. Just curious.

Reply:
I have examined the performance impact with timelt tool.

For a sentence with 100 words and each word with average 5 letters (English words length average is 5), it takes around 0.15s on average to process.
`[0.15620222315192223, 0.13365708105266094, 0.14573016110807657]`

For a sentence with 1000 words and each word with average 5 letters, it takes around 0.74s on average to process.
`[0.7371757868677378, 0.7388564720749855, 0.7497725700959563]`

Although the performance increases with the length of sentence, considering the normal use cases (user rarely sends sentence over 1000 words) I conclude the concern is rather small.